pkg/controller/backup/backup_controller.go

```diff
@@ -259,41 +259,41 @@ func (c *Controller) enqueueBackup(obj interface{}) {
 
 // detectBackupJobOrPodFailure detect backup job or pod failure.
 // it will record failure info to backup status, it is to realize reentrant logic for spec.backoffRetryPolicy. so it only record snapshot backup failed now.
-func (c *Controller) detectBackupJobOrPodFailure(backup *v1alpha1.Backup) (bool, string, string, error) {
+func (c *Controller) detectBackupJobOrPodFailure(backup *v1alpha1.Backup) (
+	podOrJobFailed bool, reason string, originalReason string, err error) {
 	var (
-		ns               = backup.GetNamespace()
-		name             = backup.GetName()
-		isPodOrJobFailed bool
-		reason           string
-		originalReason   string
-		err              error
+		ns   = backup.GetNamespace()
+		name = backup.GetName()
 	)
 
 	// if failure was recorded, get reason from backup status
 	if c.isFailureAlreadyRecorded(backup) {
 		reason = backup.Status.BackoffRetryStatus[len(backup.Status.BackoffRetryStatus)-1].RetryReason
 		originalReason = backup.Status.BackoffRetryStatus[len(backup.Status.BackoffRetryStatus)-1].OriginalReason
-		return true, reason, originalReason, nil
+		podOrJobFailed = true
+		return
 	}
 
 	// check whether backup pod and job failed
-	isPodOrJobFailed, reason, originalReason, err = c.isBackupPodOrJobFailed(backup)
+	podOrJobFailed, reason, originalReason, err = c.isBackupPodOrJobFailed(backup)
 	if err != nil {
 		klog.Errorf("Fail to check backup %s/%s pod or job status, %v", ns, name, err)
-		return false, "", "", err
+		podOrJobFailed = false
+		return
```

Comment:
I prefer old style return, it's more directly

Reply:
IMO, it's actually the opposite. For example, the previous return statement was like this: `false, "", "", err`.

who knows what is `""` or false?